File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F09.P1.go
package main

import (
	"fmt"
	"strings"
)

func probe() bool {
	s1 := strings.Repeat("ab", 2)
	s2 := strings.Repeat("ab", 2)
	// BEGIN PROBE F09.P1
	eq := s1 == s2
	// END PROBE F09.P1
	return eq
}

func main() {
	fmt.Println(probe())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F09.P1.zig
const std = @import("std");

var src: i32 = 1;

fn probe(allocator: std.mem.Allocator) !bool {
    const s1 = try std.fmt.allocPrint(allocator, "ab{d}", .{src});
    const s2 = try std.fmt.allocPrint(allocator, "ab{d}", .{src});
// BEGIN PROBE F09.P1
const eq = std.mem.eql(u8, s1, s2);
// END PROBE F09.P1
    return eq;
}

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{}\n", .{try probe(std.heap.page_allocator)});
    try out.interface.flush();
}
